### PR TITLE
Fix scheduler not logging reason denying reservation

### PR DESCRIPTION
**Pull request by vstumpf**
#### Bug/feature Description
* The scheduler would not log the reason why a reservation was denied; it would just say "Failed to confirm reservation".

#### Affected Platform(s)
* All

#### Cause / Analysis / Design
* The wrong buffer was being written into the buffer.

#### Solution Description
* Print `logmsg` instead of `errmsg` into `msgbuf`.

#### Testing logs/output
[before-fix.txt](https://github.com/PBSPro/pbspro/files/2700639/before-fix.txt)
[after-fix.txt](https://github.com/PBSPro/pbspro/files/2700640/after-fix.txt)

#### Checklist:
- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/scheduler/resv_info.c [hunk 1]
```diff
@@ -1570,7 +1570,7 @@ confirm_reservation(status *policy, int pbs_sd, resource_resv *unconf_resv, serv
 			errmsg = "";
 
 		if (rconf == RESV_CONFIRM_FAIL)
-			pbs_asprintf(&msgbuf, "PBS Failed to confirm resv: %s", errmsg);
+			pbs_asprintf(&msgbuf, "PBS Failed to confirm resv: %s", logmsg);
```
Comment (mike0042) [inline src/scheduler/resv_info.c]: Why was this changed to logmsg when errmsg was assigned to the return value of pbs_geterrmsg()?

Comment (bhroam) [reply src/scheduler/resv_info.c]: There are two reasons rconf can equal RESV_CONFIRM_FAIL.  The first is if the scheduler tried to find space for the reservation and couldn't.  The second is if the scheduler found space, but the server rejected the pbs_confirmresv() call.  pbs_geterrmsg() only will return something in the second case.  In the first case, we will print the reason the resv couldn't fit in logmsg.

Comment (mike0042) [reply src/scheduler/resv_info.c]: Thanks @bhroam. A comment might be helpful in the code. Since we only use errmsg in the else block, perhaps we should set it there rather than before the conditional. Otherwise, it is a bit misleading.

Comment (vstumpf) [reply src/scheduler/resv_info.c]: I've moved setting the error message inside the else block, as well as adding a comment in the code.

**Comment by vstumpf** [comment]
I've rebased, can someone take this forward when the checks are done?
Thanks
